File: elemi_programozasi_tetelek/kivalogatas/kivalogatas.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<double> Kivalogat(vector<double>& lista)
{
    vector<double> result;
    for (double& elem : lista)
    {
        if (0<elem)
        {
            result.push_back(elem);
        }
    }
    return result;
}

int main()
{
    vector<double> v = { -4, 5, -1, -2, 3 };
    for (auto& elem : Kivalogat(v))
    {
        cout << elem << " ";
    }
}

File: matrix/kivalogatas/kivalogatas.py
def kivalogat(lista):
    result = []
    for elem in lista:
        if 0<elem:
            result.append(elem)
    return result

l = [-4,5,-1,-3,2]

print(kivalogat(l))
# bár ezt pythonban list comprehension-nel szoktuk.
print([x for x in l if 0<x])
